src/server/client.cpp

```diff
@@ -166,13 +178,27 @@ namespace aperf {
         f.close();
       };
 
+      auto save_string = [](std::string path, std::string output) {
+        std::ofstream f;
+        f.open(path);
+        f << output << std::endl;
+        f.close();
+      };
+
+
+
       std::shared_future<void> futures[final_output.size() + 1];
 
       futures[0] = std::async(save, processed_path / "metadata.json",
                               &metadata);
 
       int future_index = 1;
 
+      futures[1] = std::async(save_string, processed_path / "external_metric_data.csv",
+                              external_metrics);
+
+      future_index++;
+
```

Comment:
I would move ```int future_index = 1``` after the ```futures[1]``` assignment and change the initial value of ```future_index``` to 2. By doing this, you won't need the extra ```future_index++```.

Reply:
Done :+1: